### ISSUE TITLE
Develop a REST API service to update a product in the shopcart

### ISSUE THREAD

**Issue by zqf0722** [2022-DevOps-Shopcarts/shopcarts#27]
**As a** User
**I need** a service to update the quantity and price of that product in the shopcart
**So that** The shopcart can reflect the price changes and amount changes happen to the product

**Assumptions**

- The url of the service is `/shopcarts/<id>/update`
- The method of this service is `PUT`
- The parameters of this method are `productName, productID, productQuantity, productPrice, and purchaseTime`
- The parameters are not required. If any of those parameters are missing, it means no changes to them
- Url should return whether the operation is successful or not
- productID should be the key to query the product
- The product quantity can be either negative or positive, if negative, it means subtract such amount of product from the shopcart, if positive, vice versa
- The negative quantity should be smaller than the quantity. Otherwise it should return error code

**Acceptance Criteria:**

```gherkin
Given the service is available
When I call the URL `PUT /shopcarts/<id>/update?productName=<name_of_product>&productID=<id_of_product>&productQuantity=<quantity_of_product>&productPrice=<price_of_each_product>&purchaseTime=<purchase_time>`
Then the product information in the shopcart should be changed accordingly if the transaction succeed
Or it should return corresponding error code if the operation fails
```

**Comment by zqf0722** [comment]
This issue is originally created by @Yuhao-Fan. Back then we decided that this issue is duplicated and got deleted, but after re-consideration, it's necessary and we open it again.

**Comment by zqf0722** [comment]
We don't need this issue any more since the reuqirement has been changed by the stakeholder.

**Comment by zqf0722** [comment]
The functionality is addressed in #32. This issue is closed because the old design is deprecated.
